platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -342,13 +342,20 @@ public State getState() {
     }
 
     @Override
+    @Deprecated
     public boolean isVisible() {
+        DeprecationLogger.deprecateMethod(Configuration.class, "isVisible")
+            .willBeRemovedInGradle10()
+            .withUpgradeGuideSection(9, "deprecate-visible-property")
+            .nagUser();
         return visible;
     }
 
     @Override
+    @Deprecated
     public Configuration setVisible(boolean visible) {
         validateMutation(MutationType.BASIC_STATE);
+        // TODO: Create a deprecation warning once https://youtrack.jetbrains.com/issue/KT-78754 is resolved
```

Comment:
Can we write up an issue for this and put it on 9.x so we make sure we don't forget about this?